src/Serilog.Sinks.Map/MapLoggerConfigurationExtensions.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Serilog.Sinks.Seq Copyright 2017 Serilog Contributors
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using Serilog.Configuration;
+using Serilog.Core;
+using Serilog.Events;
+using Serilog.Sinks.Map;
+
+namespace Serilog
+{
+    /// <summary>
+    /// Extends Serilog configuration with methods for selecting sink instances base on a log event property.
+    /// </summary>
+    public static class MapLoggerConfigurationExtensions
+    {
+        /// <summary>
+        /// Dispatch log events to a set of sinks keyed on a log event property.
+        /// </summary>
+        /// <param name="loggerSinkConfiguration">The logger sink configuration.</param>
+        /// <param name="keyPropertyName">The name of a scalar-valued property to use as a sink selector.</param>
+        /// <param name="configure">An action to configure the target sink given a key property value.</param>
+        /// <param name="defaultKey">The key property value to use when no appropriate value is attached to the log event.</param>
+        /// <param name="sinkMapCountLimit">Limits the number of sinks that will be held open concurrently within the map.
+        /// The default is to let the map grow unbounded; smaller numbers will cause sinks to be evicted when the limit is
+        /// exceeded. To keep no sinks open, zero may be specified.</param>
+        /// <param name="restrictedToMinimumLevel">The minimum log event level required 
+        /// in order to write an event to the sink.</param>
+        /// <param name="levelSwitch">A level switch to dynamically select the minimum level for events passed to the  sink.</param>
+        /// <returns>Logger configuration, allowing configuration to continue.</returns>
+        /// <exception cref="ArgumentNullException">A required parameter is null.</exception>
+        public static LoggerConfiguration Map(
+            this LoggerSinkConfiguration loggerSinkConfiguration,
+            string keyPropertyName,
+            Action<string, LoggerSinkConfiguration> configure,
+            string defaultKey = null,
+            int? sinkMapCountLimit = null,
+            LogEventLevel restrictedToMinimumLevel = LevelAlias.Minimum,
+            LoggingLevelSwitch levelSwitch = null)
+        {
+            return Map<string>(loggerSinkConfiguration, keyPropertyName, configure, defaultKey,
+                               sinkMapCountLimit, restrictedToMinimumLevel, levelSwitch);
+        }
```

Comment:
Can this non-generic `Map` overload be less restrictive on type of the `key` (check if key is `ScalarValue` and get `.ToString()` of that) for a better usability? Leaving `Map<string>` version for a strict check.
 I'm just playing with this example and it's not very intuitive:
```csharp
new LoggerConfiguration()
    .WriteTo.Map("OrderId", (_, lc) => lc.File("orders.logs"))
    .CreateLogger()
    .Information("I with {OrderId}", 42); // won't work
```

Reply:
That's a great point, will implement 👍